controllers/SiteController.php

```diff
@@ -93,6 +95,76 @@ public function actionRedirect($url)
         throw new NotFoundHttpException('The requested page was not found.');
     }
 
+    /**
+     * This action redirects old urls to the new location.
+     *
+     * There are the following types of URLs:
+     *
+     * - link to a forum/category:
+     *   https://www.yiiframework.com/forum/index.php/forum/7-framework-news/
+     *   https://www.yiiframework.com/forum/index.php/forum/7-framework-news/page__prune_day__100__sort_by__Z-A__sort_key__last_post__topicfilter__all__st__30
+     * - link to a topic:
+     *   Topic 1st page: https://www.yiiframework.com/forum/index.php/topic/3253-forum-system-updated/
+     *   Topic 2nd page: https://www.yiiframework.com/forum/index.php/topic/3253-forum-system-updated/page__st__20
+     *   Link to a post: https://www.yiiframework.com/forum/index.php/topic/3253-forum-system-updated/page__view__findpost__p__24772
+     * - link to members:
+     *   https://www.yiiframework.com/forum/index.php/user/members/
+     *   https://www.yiiframework.com/forum/index.php/user/5951-cebe/
+     * - forum rss feed:
+     *   https://www.yiiframework.com/forum/index.php/rss/forums/1-yii-framework-forum/
+     *   Redirect to topic rss
+     *
+     * Some URLs seem to be of the following form:
+     *
+     * https://www.yiiframework.com/forum/index.php?/user/5951-cebe/
+     * https://www.yiiframework.com/forum/index.php?/topic/7366-incorrect-links-to-forum-from-yii-docs/
+     *
+     *
+     */
+    public function actionRedirectForum($url = null)
+    {
+        // url must end with /
+        $forumUrl = 'https://discourse.yii.c.cebe.net/';
+
+        if ($url === 'index.php' && count($_GET) > 1) {
+            // https://www.yiiframework.com/forum/index.php?showuser=5951
+            if (isset($_GET['showuser'])) {
+                $user = User::find()->active()->where(['forum_id' => (int) $_GET['showuser']])->one();
+                if ($user !== null) {
+                    $url = "user/{$user->forum_id}-{$user->username}";
+                }
+            } else {
+                // find URLs like /forum/index.php?/user/5951-cebe/
+                foreach($_GET as $key => $value) {
+                    if ($key !== 'url' && empty($value)) {
+                        $url = ltrim($key, '/');
+                    }
+                }
+            }
+        }
+
+        if (empty($url) || $url === 'index.php' || $url === 'index.php/') {
+            return $this->redirect($forumUrl);
+        }
+
+        // prevent injection of external URLs
+        if (!Url::isRelative($url)) {
+            throw new NotFoundHttpException('The requested page was not found.');
```

Comment:
how about to just redirect to homepage?

Reply:
this is a security check to not redirect to external Urls if url is manipulated, this should never happen with real URLs.